File: c_plus/Excel_Sheet_Column_Number.cpp
#include<iostream>
#include<string>
using namespace std;

class Solution {
public:
    int titleToNumber(string columnTitle) {
        int res = 0;
        for (int i = 0; i < columnTitle.size(); i++) {
            res = res * 26 + (long)columnTitle[i] - 64;
        }
        return res;
    }
};

int main() {
    Solution s;
    cout << s.titleToNumber("AB") << endl;
}

File: py/Excel_Sheet_Column_Number.py
class Solution:
    def titleToNumber(self, columnTitle):
        ## method 1
        result = 0
        n = len(columnTitle)
        for i in range(n):
            result = result * 26 + ord(columnTitle[i]) - 64
        return result

        ## method 2
        # if len(columnTitle) == 1:
        #     return ord(columnTitle)-64
        # return ord(columnTitle[-1]) - 64 + 26 * \
        #         self.titleToNumber(columnTitle[:-1]) 